cmd/experimental/migrate/aws/main.go

```diff
@@ -0,0 +1,165 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// aws-migrate is a command-line tool for migrating data from a tlog-tiles
+// compliant log, into a Tessera log instance hosted on AWS.
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	aaws "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/go-sql-driver/mysql"
+	tessera "github.com/transparency-dev/trillian-tessera"
+	"github.com/transparency-dev/trillian-tessera/client"
+	"github.com/transparency-dev/trillian-tessera/storage/aws"
+	"k8s.io/klog/v2"
+)
+
+var (
+	bucket            = flag.String("bucket", "", "Bucket to use for storing log")
+	dbName            = flag.String("db_name", "", "AuroraDB name")
+	dbHost            = flag.String("db_host", "", "AuroraDB host")
+	dbPort            = flag.Int("db_port", 3306, "AuroraDB port")
+	dbUser            = flag.String("db_user", "", "AuroraDB user")
+	dbPassword        = flag.String("db_password", "", "AuroraDB user")
+	dbMaxConns        = flag.Int("db_max_conns", 0, "Maximum connections to the database, defaults to 0, i.e unlimited")
+	dbMaxIdle         = flag.Int("db_max_idle_conns", 2, "Maximum idle database connections in the connection pool, defaults to 2")
+	s3Endpoint        = flag.String("s3_endpoint", "", "Endpoint for custom non-AWS S3 service")
+	s3AccessKeyID     = flag.String("s3_access_key", "", "Access key ID for custom non-AWS S3 service")
+	s3SecretAccessKey = flag.String("s3_secret", "", "Secret access key for custom non-AWS S3 service")
+
+	sourceURL  = flag.String("source_url", "", "Base URL for the source log.")
+	numWorkers = flag.Uint("num_workers", 30, "Number of migration worker goroutines.")
+)
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+	ctx := context.Background()
+
+	if len(*sourceURL) == 0 {
+		klog.Exit("Missing parameter: --source_url")
+	}
+	srcURL, err := url.Parse(*sourceURL)
+	if err != nil {
+		klog.Exitf("Invalid --source_url %q: %v", *sourceURL, err)
+	}
+	src, err := client.NewHTTPFetcher(srcURL, nil)
+	if err != nil {
+		klog.Exitf("Failed to create HTTP fetcher: %v", err)
+	}
+	sourceCP, err := src.ReadCheckpoint(ctx)
+	if err != nil {
+		klog.Exitf("fetch initial source checkpoint: %v", err)
+	}
+	bits := strings.Split(string(sourceCP), "\n")
+	sourceSize, err := strconv.ParseUint(bits[1], 10, 64)
+	if err != nil {
+		klog.Exitf("invalid CP size %q: %v", bits[1], err)
+	}
+	sourceRoot, err := base64.StdEncoding.DecodeString(bits[2])
+	if err != nil {
+		klog.Exitf("invalid checkpoint roothash %q: %v", bits[2], err)
+	}
```

Comment:
I know this came from the GCP code, but we should probably have an issue to make all these migrate cmds do this safely/properly.

Reply:
Replaced with existing unsafe parse function and added TODO.